doped/utils/qehacks.py

```
@@ -0,0 +1,56 @@
+
+        def setter(self, value):
+            setattr(self, backing_attr, value)
+
+        setattr(cls, prop_name, property(getter, setter))
+
+    @classmethod
+    def patch_pwxml_properties(cls):
+        """
+        Applies all useful patches to PWxml so these attributes become writable.
+        """
+        from pymatgen.io.espresso.outputs import PWxml
```

Ruff (0.12.2)

`import` should be at the top-level of a file

(PLC0415)